### DISCUSSION TITLE
gym wrapper, train_muzero_with_gym_env

### DISCUSSION THREAD

**Discussion by rosnu** [opendilab/LightZero#280] [General]
Hello,
1.
Could someone verify if entry_type == "train_muzero_with_gym_env" (setting at the end of the script) is working? 
I tried it with:
python3 -u zoo/classic_control/cartpole/config/cartpole_efficientzero_config.py

Error:
/content/LightZero/lzero/worker/muzero_collector.py in collect(self, n_episode, train_iter, policy_kwargs, collect_with_pure_policy)
    329             policy_kwargs = {}
    330         temperature = policy_kwargs['temperature']
**--> 331         epsilon = policy_kwargs['epsilon']**
    332 
    333         collected_episode = 0
KeyError: 'epsilon'

entry_type == "train_muzero" seems to be working fine.
I'm more of an "enthusiast" than a professional, so I would prefer to continue with a working example.

2.
Is it beneficial to build a custom environment from scratch according to the LiteZero schema compared to using a Gym/Gymnasium environment and a wrapper? I can't find any configuration example for any train_xxxx_with_gym_env in the zoo.

Thank you!

**Comment by puyuan1996** [comment]
Hello, regarding the first issue you mentioned, we have already fixed it in this [commit](https://github.com/opendilab/LightZero/commit/f1511fb5cdda4d31e61c5831c877d36215b49b22). As for the second issue, you can refer to our documentation: [Customizing Environments Tutorial](https://opendilab.github.io/LightZero/tutorials/envs/customize_envs.html). If you encounter any problems while customizing the environment, feel free to start a discussion here. Thank you again for your attention and support!

**Reply by r-wedeen** [reply]
Hi, I am also using train_muzero_with_gym_env to wrap my gymnasium environment with LightZeroEnvWrapper. What should my create_config look like? I was not able to find this information in the Customizing Environments Tutorial mentioned above nor in the Customizing Configuration Files Tutorial. Thanks!

**Reply by puyuan1996** [reply]
Hello, if your environment has been correctly configured according to the documentation, the `create_config` should be similar to the configuration in this [link](https://github.com/opendilab/LightZero/blob/main/zoo/classic_control/cartpole/config/cartpole_muzero_config.py#L77). Of course, you may need to add extra parameters for the custom environment in `env` filed of the `main_config`. If you have any questions, feel free to ask at any time. Thank you for your support!
